lib/oauth2.js

```diff
@@ -33,7 +33,8 @@ manager.addJob('getCode', {
 		
 		winston.info('Run job oauth2:getCode');
 		
-		var oaData = GLOBAL.config.authentication[api.auth.provider];
+		var oaData = api.auth.provider;
+		_.isObject(oaData) || (oaData = GLOBAL.config.authentication[oaData]);
```

Comment:
This is the only significant change (same as line 72), the rest are just re-indentation.